### PR TITLE
Anurag/22

### LINKED ISSUE #22: Create a button acknowledging the information and adding a point

### PR DESCRIPTION
**Pull request by AnuragVinodh**
close #22

### CODE REVIEW

**Comment by jbennyhuang** [comment]
you should rebase to main

File: .backend/src/controllers/user.ts [hunk 1]
```diff
@@ -1,8 +1,109 @@
...
+import User from "../models/user";
+import { z } from "zod";
 
 export default class UserController {
+
+  static async incrementByOne(request: HttpRequest, context: InvocationContext) {
+    try {
+      const token = await extractToken(request);
+
+      // Token Check -----------------------------------------------
```
Comment (jbennyhuang) [inline .backend/src/controllers/user.ts]: remove

File: .backend/src/functions/increment-points.ts
Comment (jbennyhuang) [inline .backend/src/functions/increment-points.ts:1]: put this function in `functions/users.ts` and convert to a PATCH request

File: .backend/src/models/user.ts [hunk 2]
```diff
@@ -37,4 +42,32 @@ export default class User implements IUser {
 
     return Some(new User(user.resource));
   }
+
+  static async updatePoints(id: string, value: number): Promise<ReturnObj> {
```
Comment (jbennyhuang) [inline .backend/src/models/user.ts]: remove `static` and `id`

Comment (jbennyhuang) [reply .backend/src/models/user.ts]: use the monads

File: .backend/src/controllers/user.ts (hunk 1 above)
Comment (AnuragVinodh) [reply .backend/src/controllers/user.ts]: removed comments
added a new extractor for ID 
refactored code

**Comment by AnuragVinodh** [comment]
> you should rebase to main

I rebased right after you pushed the code with the extractors code. did you push after that? I think the rebase is complete and worked fine

File: .backend/src/models/user.ts (hunk 2 above)
Comment (AnuragVinodh) [reply .backend/src/models/user.ts]: removed static and I am using Ok and Err as return values for User schema methods

Comment (AnuragVinodh) [reply .backend/src/models/user.ts]: why do I need to remove id? I am using it to get the item

**Review by jbennyhuang — APPROVED** [review]

**Pull request merged**